addons/product/models/product_template.py

```diff
@@ -602,6 +602,23 @@ def _create_variant_ids(self):
                 variants_to_create += current_variants_to_create
                 variants_to_activate += current_variants_to_activate
 
+            else:
+                all_variants_dict = {}
+                for variant in all_variants:
+                    variant_entry = all_variants_dict.get(variant.combination_indices)
```

Comment:
Note: `combination_indices` is a "hack" that should not be used outside of unique index and `_get_variant_id_for_combination`